### PR TITLE
Change mixer_codegen.sh to use protoc dockerfile

### PR DESCRIPTION
**Pull request by gargnupur**
This is starting the work discussed in https://github.com/istio/istio/pull/6606/files

### CODE REVIEW

File: bin/counterfeiter.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+WD=$(dirname $0)
+WD=$(cd $WD; pwd)
+ROOT=$(dirname $WD)
```
Comment (ozevren) [inline bin/counterfeiter.sh]: FYI, you can use:
```DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"```
To figure out the directory of the script itself. You can calculate the path of istio.io from there.

Comment (gargnupur) [reply bin/counterfeiter.sh]: Thanks.. Updated

**Comment by gargnupur** [comment]
@geeknoid , @ozevren , @mandarjog , @douglas-reid : all the tests passed, can you guys please take a look at this again for lgtm?

File: docker/build_and_push_go_generate_dependency.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+# This is a temporary script, only to be used until we have a better official
```
Comment (ozevren) [inline docker/build_and_push_go_generate_dependency.sh]: Can you create an issue for this and add a TODO here pointing to the issue? This way, we can track this and backtrack to the appropriate location.

File: docker/Dockerfile.go_generate_dependency [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM alpine:3.7 as go_generate_dependency_builder
+RUN apk add --no-cache build-base git
+
+RUN apk update
+RUN apk add --no-cache go>1.10
+ENV GOPATH=/go \
+    PATH=/go/bin/:$PATH
+RUN go get -u -v \
+    github.com/jteeuwen/go-bindata/... \
```
Comment (ozevren) [inline docker/Dockerfile.go_generate_dependency]: Do we need to peg the version numbers of these as well?

Comment (gargnupur) [reply docker/Dockerfile.go_generate_dependency]: Don't think they are actively pushing releases.. go-bindata last release was 2014 and for counterfeiter 2017, thus using from head seems better as we would anyways bind to a good know tried version using the dockerfile
What do you think?

File: docker/build_and_push_go_generate_dependency.sh (hunk 2 above)
Comment (gargnupur) [reply docker/build_and_push_go_generate_dependency.sh]: Thanks.. Added

**Pull request merged**